client/src/containers/NotRegisteredPage.tsx

```diff
@@ -56,132 +62,139 @@ export default class NotRegisteredPage extends Component<Props, State> {
-                  >
-                    <Trans render="u">Building Classification</Trans>
-                  </a>
-                  : {buildingInfo.bldgclass}
-                </i>
-                )
-              </Nobr>
-            </p>
-          </h6>
-        );
-        break;
-    }
-
     const usersInputAddressFragment = usersInputAddress ? (
       <>
         {usersInputAddress.housenumber === " " ? "" : usersInputAddress.housenumber + " "}
         {usersInputAddress.streetname !== " " && usersInputAddress.streetname}
       </>
     ) : null;
 
+    const registrationMissingOrExpired =
+      !buildingInfo.registrationenddate ||
+      getTodaysDate() > new Date(buildingInfo.registrationenddate);
+
+    let registrationMessageText;
+
+    if (!usersInputAddress) {
+      registrationMessageText = <Trans>No registration found.</Trans>;
+    } else if (buildingInfo.registrationenddate) {
+      registrationMessageText = (
+        <Trans>Incomplete registration for {usersInputAddressFragment}.</Trans>
+      );
+    } else {
+      registrationMessageText = (
+        <Trans>No registration found for {usersInputAddressFragment}.</Trans>
+      );
+    }
+
+    let buildingTypeMessageText;
+
+    if (buildingInfo.unitsres === 0) {
+      buildingTypeMessageText = (
+        <Trans>
+          This property doesn't have any residential units, so it is not required to register with
+          HPD.
+        </Trans>
+      );
+    } else if (buildingInfo.unitsres < 3) {
+      buildingTypeMessageText = (
+        <Trans>
+          This property has fewer than 3 residential units. If the landlord doesn't reside there, it
+          should be registered with HPD.
+        </Trans>
+      );
+    } else if (buildingInfo.unitsres >= 3) {
+      buildingTypeMessageText = (
+        <Trans>
+          This property has more than 2 residential units, so it should be registered with HPD.
+        </Trans>
+      );
+    }
+
+    const formattedLastRegDate = Helpers.formatDate(
+      buildingInfo.lastregistrationdate,
+      longDateOptions,
+      locale
+    );
+
+    const formattedRegEndDate = Helpers.formatDate(
+      buildingInfo.registrationenddate,
+      longDateOptions,
+      locale
+    );
+
+    const lastRegisteredDates = buildingInfo.registrationenddate ? (
+      <div className="card-body-registration text-center">
+        {buildingInfo.lastregistrationdate ? (
+          <p>
+            <b>
+              <Trans>Last registered:</Trans>
+            </b>{" "}
+            {formattedLastRegDate}
+            <span className="text-danger">
+              {" "}
+              <Trans>(expired {formattedRegEndDate})</Trans>
+            </span>
+          </p>
+        ) : (
+          <p>
+            <b>
+              <Trans>Registration expired:</Trans>
+            </b>{" "}
+            <span className="text-danger">
+              {" "}
+              <Trans>{formattedRegEndDate}</Trans>
```

Comment:
@austensen just realized, this date doesn't need to be internationalized since there is no text beyond just the date!

Reply:
oh good catch, thanks, just fixed this